diff --git a/mkspecs/common/macx.conf b/mkspecs/common/macx.conf
index d16b77acb8..adab89d673 100644
--- a/mkspecs/common/macx.conf
+++ b/mkspecs/common/macx.conf
@@ -5,14 +5,11 @@
 QMAKE_PLATFORM         += macos osx macx
 QMAKE_MAC_SDK           = macosx
 
-QMAKE_MACOSX_DEPLOYMENT_TARGET = 10.13
-QMAKE_APPLE_DEVICE_ARCHS = x86_64
+QMAKE_MACOSX_DEPLOYMENT_TARGET = 11.00
+QMAKE_APPLE_DEVICE_ARCHS = arm64
 
-# Should be 10.15, but as long as the CI builds with
-# older SDKs we have to keep this.
-QT_MAC_SDK_VERSION_MIN = 10.14
-
-QT_MAC_SDK_VERSION_MAX = 10.15
+QT_MAC_SDK_VERSION_MIN = 11.00
+QT_MAC_SDK_VERSION_MAX = 11.00
 
 device.sdk = macosx
 device.target = device
