diff --git a/mkspecs/common/macx.conf b/mkspecs/common/macx.conf
index 61a1300860..117bd50d5e 100644
--- a/mkspecs/common/macx.conf
+++ b/mkspecs/common/macx.conf
@@ -5,13 +5,10 @@
 QMAKE_PLATFORM         += macos osx macx
 QMAKE_MAC_SDK           = macosx
 
-QMAKE_MACOSX_DEPLOYMENT_TARGET = 10.13
-QMAKE_APPLE_DEVICE_ARCHS = x86_64
-
-# Should be 10.15, but as long as the CI builds with
-# older SDKs we have to keep this.
-QT_MAC_SDK_VERSION_MIN = 10.14
+QMAKE_MACOSX_DEPLOYMENT_TARGET = 11.0
+QMAKE_APPLE_DEVICE_ARCHS = arm64
 
+QT_MAC_SDK_VERSION_MIN = 11.0
 QT_MAC_SDK_VERSION_MAX = 11.0
 
 device.sdk = macosx
